extensions/functions/src/main/java/io/crate/window/OffsetValueFunctions.java

```diff
@@ -46,7 +46,62 @@
  */
 public class OffsetValueFunctions implements WindowFunction {
 
-    private abstract class OffsetDirection {
+    private enum OffsetDirection {
+        FORWARD {
```

Comment:
I initially created inherited Forward/Backward Direction classes as suggested, but this way the instances could be re-used. Please let me know what you think.

Reply:
I'm not sure if this will be safe, given that `cachedNonNullIndex` is a local property within the enum. If two window functions run concurrently and both access the same `OffsetDirection` instance, they'll both operate on the same property and mutate it.